app/controllers/api/projects.php

```diff
@@ -223,73 +224,74 @@
         $sharedTables = $sharedTablesV1 || $sharedTablesV2;
 
         if (!$sharedTablesV2) {
-            $adapter = $pools->get($dsn->getHost())->pop()->getResource();
-            $dbForProject = new Database($adapter, $cache);
-
-            if ($sharedTables) {
-                $dbForProject
-                    ->setSharedTables(true)
-                    ->setTenant($sharedTablesV1 ? $project->getInternalId() : null)
-                    ->setNamespace($dsn->getParam('namespace'));
-            } else {
-                $dbForProject
-                    ->setSharedTables(false)
-                    ->setTenant(null)
-                    ->setNamespace('_' . $project->getInternalId());
-            }
-
-            $create = true;
-
-            try {
-                $dbForProject->create();
-            } catch (Duplicate) {
-                $create = false;
-            }
+            $pools->get($dsn->getHost())->use(function (Adapter $adapter) use ($project, $dsn, $cache, $projectTables, $sharedTables, $sharedTablesV1) {
```

Comment:
Why do we need this wrapper syntax? Can we just call the database and let this be handled under the hood?

Reply:
Yep updated